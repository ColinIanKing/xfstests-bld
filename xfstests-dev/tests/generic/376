#! /bin/bash
# FSQA Test No. 376
#
# Test that if we rename a file, without changing its parent directory, create
# a new file that has the old name of the file we renamed, doing an fsync
# against the file we renamed works correctly and after a power failure both
# files exists.
#
#-----------------------------------------------------------------------
#
# Copyright (C) 2016 SUSE Linux Products GmbH. All Rights Reserved.
# Author: Filipe Manana <fdmanana@suse.com>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#-----------------------------------------------------------------------
#

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"
tmp=/tmp/$$
status=1	# failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 15

_cleanup()
{
	_cleanup_flakey
	cd /
	rm -f $tmp.*
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter
. ./common/dmflakey

# real QA test starts here
_supported_fs generic
_supported_os Linux
_require_scratch
_require_dm_target flakey
_require_metadata_journaling $SCRATCH_DEV

rm -f $seqres.full

_scratch_mkfs >>$seqres.full 2>&1
_init_flakey
_mount_flakey

# Create our test directories and files.
mkdir $SCRATCH_MNT/dir
touch $SCRATCH_MNT/dir/foo
# Make sure everything is durably persisted.
sync
mv $SCRATCH_MNT/dir/foo $SCRATCH_MNT/dir/bar
touch $SCRATCH_MNT/dir/foo
$XFS_IO_PROG -c "fsync" $SCRATCH_MNT/dir/bar

_flakey_drop_and_remount

echo "Filesystem contents after log replay:"
ls -R $SCRATCH_MNT/dir | _filter_scratch

_unmount_flakey
status=0
exit
