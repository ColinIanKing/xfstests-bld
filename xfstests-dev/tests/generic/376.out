QA output created by 376
Filesystem contents after log replay:
SCRATCH_MNT/dir:
bar
foo
